api/definitions/definitions.proto

```diff
@@ -0,0 +1,895 @@
+syntax = "proto3";
+
+package definitions;
+
+option go_package = "github.com/talos-systems/talos/pkg/definitions";
+
+import "common/common.proto";
+import "google/protobuf/timestamp.proto";
+
+// AffiliateSpec describes Affiliate state.
+message ClusterAffiliateSpec {
+  string node_id = 1;
+  repeated common.NetIP addresses = 2;
+  string hostname = 3;
+  string nodename = 4;
+  string operating_system = 5;
+  int64 machine_type = 6;
```

Comment:
this is an interesting one... as it should be an `enum` type actually, and things like this one which are constants in Go with enumeration

we can create it as `message`-local enum to avoid dependencies, something like:

```proto
message A {
   enum B {
      A = 0;
      B = 1;
   }
   
   B enum_field = 3;
}
```